std/algorithm/iteration.d

```diff
@@ -1113,20 +1115,30 @@ public:
     arr.each!"a++";
     assert(arr == [2, 3, 4, 5, 6, 2]);
 
+    auto m = arr.map!(n => n);
     // by-ref lambdas are not allowed for non-ref ranges
-    static assert(!is(typeof(arr.map!(n => n).each!((ref n) => n++))));
+    static assert(!__traits(compiles, m.each!((ref n) => n++)));
 
     // The default predicate consumes the range
-    auto m = arr.map!(n => n);
     (&m).each();
     assert(m.empty);
+}
+
+/// `each` can pass an index variable for iterable objects which support this
+@safe unittest
+{
+    auto arr = new int[4];
 
-    // Indexes are also available for in-place mutations
-    arr[] = 0;
     arr.each!"a=i"();
```

Comment:
This is breaking the test suite, arr was long before. Will fix a bit later today.